### PR TITLE
[PLE-866] refactor(tf-workflow): set and consistently use TF work dir

### PR DESCRIPTION
**Pull request by gustavclausen**
With these changes, the working directory for Terraform is set and used consistently in all steps in the workflow.

The primary reason to introduce this refactor is to ensure that the procedure to save the "error state" (if the Terraform state update fails) does not depend on the work directory being set in the 'init' step that only runs if there's a cache miss for the Terraform dependencies.

### CODE REVIEW

**Review by vyrwu — APPROVED** [review]

File: .github/workflows/terraform-setup.yml [hunk 1]
```diff
@@ -145,6 +145,13 @@ jobs:
           secrets: ${{ toJSON(secrets) }}
           convert: tf_case
 
+      - name: Determine TF working directory
+        id: tf-work-dir
+        shell: bash
+        run: |
+          ${{ inputs.navigate_to_terraform }}
+          echo "TF_PATH=$(pwd)" >> $GITHUB_OUTPUT
```
Comment (nicolajv) [inline .github/workflows/terraform-setup.yml:153]: We have a variable `terraform_root_path` that might be able to be replaced with this TF_PATH solution as well? I would suggest deprecating that variable and try to use your new solution

**Review by nicolajv — APPROVED** [review]
Very nice, only an optional suggestion 🔥

File: .github/workflows/terraform-setup.yml (hunk 1 above)
Comment (gustavclausen) [reply .github/workflows/terraform-setup.yml:156]: @nicolajv LMKWYT :)

Comment (nicolajv) [reply .github/workflows/terraform-setup.yml:156]: I think you did an awesome job as always 👏 🔥

**Pull request merged**